libc 0.2.105 broken as libstd dep on macOS

# Reproduce

Update libc in rust-lang/rust to version 0.2.105. Then try to compile on macOS.

# Errors

```
error[E0412]: cannot find type `Option` in this scope
   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.105/src/unix/bsd/apple/b64/x86_64/mod.rs:113:19
    |
113 |         pub size: Option<unsafe extern "C" fn(
    |                   ^^^^^^ not found in this scope
    |
help: consider importing this enum
    |
178 |         use Option;
    |

error[E0412]: cannot find type `Option` in this scope
   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.105/src/unix/bsd/apple/b64/x86_64/mod.rs:117:21
    |
117 |         pub malloc: Option<unsafe extern "C" fn(
    |                     ^^^^^^ not found in this scope
    |
help: consider importing this enum
    |
178 |         use Option;
    |

error[E0412]: cannot find type `Option` in this scope
   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.105/src/unix/bsd/apple/b64/x86_64/mod.rs:121:21
    |
164 |         pub pressure_relief: Option<unsafe extern "C" fn(
    |                              ^^^^^^ not found in this scope
    |
help: consider importing this enum
    |
178 |         use Option;
    |

error[E0412]: cannot find type `Option` in this scope
   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.105/src/unix/bsd/apple/b64/x86_64/mod.rs:168:30
    |
168 |         pub claimed_address: Option<unsafe extern "C" fn(
    |                              ^^^^^^ not found in this scope
    |
help: consider importing this enum
    |
178 |         use Option;
    |

warning: an associated function with this name may be added to the standard library in the future
 --> compiler-builtins/src/float/pow.rs:8:19
  |
8 |     let mut pow = i32::abs_diff(b, 0);
  |                   ^^^^^^^^^^^^^
  |
  = note: `#[warn(unstable_name_collisions)]` on by default
  = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
  = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
  = help: call with fully qualified syntax `int::Int::abs_diff(...)` to keep using the current method
  = help: add `#![feature(int_abs_diff)]` to the crate attributes to enable `rustc_std_workspace_core::num::<impl i32>::abs_diff`
```

# Meta

This caused cg_clif's CI to fail as I updated libc: https://github.com/bjorn3/rustc_codegen_cranelift/runs/4022217585?check_suite_focus=true

--- a/src/unix/bsd/apple/b64/x86_64/mod.rs
+++ b/src/unix/bsd/apple/b64/x86_64/mod.rs
@@ -110,62 +110,62 @@ s! {
     pub struct malloc_zone_t {
         _reserved1: *mut ::c_void,
         _reserved2: *mut ::c_void,
-        pub size: Option<unsafe extern "C" fn(
+        pub size: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             ptr: *const ::c_void,
         ) -> ::size_t>,
-        pub malloc: Option<unsafe extern "C" fn(
+        pub malloc: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             size: ::size_t,
         ) -> *mut ::c_void>,
-        pub calloc: Option<unsafe extern "C" fn(
+        pub calloc: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             num_items: ::size_t,
             size: ::size_t,
         ) -> *mut ::c_void>,
-        pub valloc: Option<unsafe extern "C" fn(
+        pub valloc: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             size: ::size_t
         ) -> *mut ::c_void>,
-        pub free: Option<unsafe extern "C" fn(
+        pub free: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             ptr: *mut ::c_void
         )>,
-        pub realloc: Option<unsafe extern "C" fn(
+        pub realloc: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             ptr: *mut ::c_void,
             size: ::size_t,
         ) -> *mut ::c_void>,
-        pub destroy: Option<unsafe extern "C" fn(zone: *mut malloc_zone_t)>,
+        pub destroy: ::Option<unsafe extern "C" fn(zone: *mut malloc_zone_t)>,
         pub zone_name: *const ::c_char,
-        pub batch_malloc: Option<unsafe extern "C" fn(
+        pub batch_malloc: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             size: ::size_t,
             results: *mut *mut ::c_void,
             num_requested: ::c_uint,
         ) -> ::c_uint>,
-        pub batch_free: Option<unsafe extern "C" fn(
+        pub batch_free: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             to_be_freed: *mut *mut ::c_void,
             num_to_be_freed: ::c_uint,
         )>,
         pub introspect: *mut malloc_introspection_t,
         pub version: ::c_uint,
-        pub memalign: Option<unsafe extern "C" fn(
+        pub memalign: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             alignment: ::size_t,
             size: ::size_t,
         ) -> *mut ::c_void>,
-        pub free_definite_size: Option<unsafe extern "C" fn(
+        pub free_definite_size: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             ptr: *mut ::c_void,
             size: ::size_t
         )>,
-        pub pressure_relief: Option<unsafe extern "C" fn(
+        pub pressure_relief: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             goal: ::size_t,
         ) -> ::size_t>,
-        pub claimed_address: Option<unsafe extern "C" fn(
+        pub claimed_address: ::Option<unsafe extern "C" fn(
             zone: *mut malloc_zone_t,
             ptr: *mut ::c_void,
         ) -> ::boolean_t>,

